File: nodejs/src/avenger/avenger.controller.ts
import {Body, Controller, Delete, Get, Param, Post, Put} from '@nestjs/common';
import {AvengerService} from './avenger.service';
import {ApiOperation, ApiTags} from "@nestjs/swagger";
import {Avenger} from "./avenger.entity";

@ApiTags('avengers')
@Controller('v1/avengers')
export class AvengerController {
    constructor(private readonly service: AvengerService) {
    }

    @Get()
    @ApiOperation({
        description: 'Find All Avengers'
    })
    async findAll(): Promise<Avenger[]> {
        return await this.service.findAll();
    }

    @Get(':id')
    async findOne(@Param('id') id: string): Promise<Avenger> {
        return await this.service.findOne(id);
    }

    @Post()
    async create(@Body() avenger: Avenger): Promise<Avenger> {
        return await this.service.create(avenger);
    }

    @Put(':id')
    async update(@Param('id') id: string, @Body() avenger: Avenger): Promise<Avenger> {
        return await this.service.update(id, avenger);
    }

    @Delete(':id')
    async delete(@Param('id') id: string): Promise<void> {
        await this.service.delete(id);
    }
}

File: java/src/main/java/io/egen/controller/AvengerController.java
package io.egen.controller;

import io.egen.entity.Avenger;
import io.egen.service.AvengerService;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/v1/avengers")
public class AvengerController {

    private AvengerService service;

    public AvengerController(AvengerService service) {
        this.service = service;
    }

    @GetMapping
    public List<Avenger> findAll() {
        return this.service.findAll();
    }

    @GetMapping("{id}")
    public Avenger findOne(@PathVariable(required = true) String id) {
        return this.service.findOne(id);
    }


    @PostMapping
    public Avenger create(@RequestBody Avenger avenger) {
        return this.service.create(avenger);
    }

    @PutMapping("{id}")
    public Avenger update(@PathVariable(required = true) String id, @RequestBody Avenger avenger) {
        return this.service.update(id, avenger);
    }


    @DeleteMapping("{id}")
    public void delete(@PathVariable(required = true) String id) {
        this.service.delete(id);
    }
}